packages/davinci-client/src/lib/davinci.api.ts

```diff
@@ -79,45 +84,44 @@ export const davinciApi = createApi({
        * parameters are pre-typed from the library.
        */
       async onQueryStarted(_, api) {
+        let response;
+
         try {
-          await api.queryFulfilled;
-        } catch (error) {
+          const query = await api.queryFulfilled;
+          response = query.meta?.response;
+        } catch (err: unknown) {
+          const error = err as ThrownQueryError;
+
           /**
            * This error is thrown when the query is rejected. We don't
            * want to do anything with it for now.
            */
+          response = error.meta?.response;
         }
```

Comment:
Getting the response differs between a success or an error, returned value versus `error` object in the `catch`.